encoding/ssz/query/path.go

```diff
@@ -32,31 +42,85 @@ func ParsePath(rawPath string) ([]PathElement, error) {
+		matches := re.FindStringSubmatch(processingField)
+		if len(matches) == 2 {
+			pathElement.Length = true
+			// Extract the inner expression between len( and ) and continue parsing on that
+			processingField = matches[1]
+		}
 
-			fieldName = parts[0]
-			indexPart := strings.TrimSuffix(parts[1], "]")
-			if indexPart == "" {
-				return nil, errors.New("index cannot be empty")
-			}
+		// Default name is the full working string (may be updated below if it contains indices)
+		pathElement.Name = processingField
 
-			indexValue, err := strconv.ParseUint(indexPart, 10, 64)
+		if strings.Contains(processingField, "[") {
+			// Split into field and indices, e.g., "array[0][1]" -> name:"array", indices:{0,1}
+			pathElement.Name = extractFieldName(processingField)
+			indices, err := extractArrayIndices(processingField)
 			if err != nil {
-				return nil, fmt.Errorf("invalid index in path element %s: %w", elem, err)
+				return []PathElement{}, err
 			}
-			index = &indexValue
+			// Although extractArrayIndices supports multiple indices,
+			// only a single index is supported per PathElement, e.g., "transactions[0]" is valid
+			// while "transactions[0][0]" is rejected explicitly.
+			if len(indices) != 1 {
+				return []PathElement{}, fmt.Errorf("multiple indices not supported in token %q", processingField)
+			}
+			pathElement.Index = &indices[0]
+
 		}
 
-		path = append(path, PathElement{Name: fieldName, Index: index})
+		path = append(path, pathElement)
 	}
 
 	return path, nil
 }
+
+// extractFieldName extracts the field name from a path element name (removes array indices)
+// For example: "field_name[5]" returns "field_name"
+func extractFieldName(name string) string {
+	if idx := strings.Index(name, "["); idx != -1 {
+		return name[:idx]
+	}
+	return strings.ToLower(name)
+}
+
+// extractArrayIndices returns every bracketed, non-negative index in the name,
+// e.g. "array[0][1]" -> []uint64{0, 1}. Errors if none are found or if any index is invalid.
+func extractArrayIndices(name string) ([]uint64, error) {
+	// Match all bracketed content, then we'll parse as unsigned to catch negatives explicitly
+	re := regexp.MustCompile(`\[\s*([^\]]+)\s*\]`)
+	matches := re.FindAllStringSubmatch(name, -1)
+
+	if len(matches) == 0 {
+		return nil, errors.New("no array indices found")
+	}
+
+	indices := make([]uint64, 0, len(matches))
+	for _, m := range matches {
+		raw := strings.TrimSpace(m[1])
+		// Forbid signs explicitly; we want a clear error similar to ParseUint's message
+		if strings.HasPrefix(raw, "-") {
+			return nil, fmt.Errorf("cannot process negative indices %q", raw)
+		}
+		idx, err := strconv.ParseUint(raw, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid array index: %w", err)
+		}
+		indices = append(indices, idx)
+	}
+	return indices, nil
+}
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+func ToSnakeCase(str string) string {
```

Comment:
```suggestion
func toSnakeCase(str string) string {
```
I'm pretty sure that we don't have to expose this function.